mux/src/tmux_commands.rs

```diff
@@ -59,148 +84,535 @@ impl TmuxDomainState {
-            let mux = Mux::get();
             let size = TerminalSize {
-                rows: pane.pane_height as usize,
-                cols: pane.pane_width as usize,
+                rows: window.window_height as usize,
+                cols: window.window_width as usize,
                 pixel_width: 0,
                 pixel_height: 0,
                 dpi: 0,
             };
 
-            let child = TmuxChild {
-                active_lock: active_lock.clone(),
-            };
+            let tab = Arc::new(Tab::new(&size));
+            tab.set_title(&format!("{}", &window.window_name));
+            mux.add_tab_no_panes(&tab);
 
-            let terminal = wezterm_term::Terminal::new(
-                size,
-                std::sync::Arc::new(config::TermConfig::new()),
-                "WezTerm",
-                config::wezterm_version(),
-                Box::new(writer.clone()),
-            );
+            let _ = self.add_attached_window(window, &tab.tab_id())?;
 
-            let local_pane: Arc<dyn Pane> = Arc::new(LocalPane::new(
-                local_pane_id,
-                terminal,
-                Box::new(child),
-                Box::new(pane_pty),
-                Box::new(writer),
-                self.domain_id,
-                "tmux pane".to_string(),
-            ));
+            let mut split_stack;
+            let mut split_direction;
 
-            let tab = Arc::new(Tab::new(&size));
-            tab.assign_pane(&local_pane);
-
-            self.create_gui_window();
-            let mut gui_window = self.gui_window.lock();
-            let gui_window_id = match gui_window.as_mut() {
-                Some(x) => x,
-                None => {
-                    anyhow::bail!("No tmux gui created");
+            let mut split_pane_index = 1;
+            for l in &window.layout {
+                match l {
+                    TmuxLayout::SinglePane(p) => {
+                        let local_pane = self.create_pane(&p).unwrap();
+                        tab.assign_pane(&local_pane);
+                        self.add_attached_pane(p.window_id, p.pane_id)?;
+                        let _ = mux.add_pane(&local_pane);
+                        break;
+                    }
+
+                    TmuxLayout::SplitHorizontal(x) => {
+                        split_direction = SplitDirection::Horizontal;
+                        split_stack = x;
+                    }
+
+                    TmuxLayout::SplitVertical(x) => {
+                        split_direction = SplitDirection::Vertical;
+                        split_stack = x;
+                    }
                 }
-            };
 
-            mux.add_tab_and_active_pane(&tab)?;
+                for p in split_stack {
+                    let local_pane;
+                    if !self.check_pane_attached(p.window_id, p.pane_id) {
+                        local_pane = self.create_pane(&p).unwrap();
```

Comment:
Can we propagate a reasonable error here instead of panic?
```suggestion
                        local_pane = self.create_pane(&p).context("failed to create pane")?;
```